sql/sem/eval: TestEval fails on s390x

**[How to gain access to an s390x machine for testing (Internal link)](https://cockroachlabs.atlassian.net/wiki/spaces/devinf/pages/4536467683/Reproducing+an+issue+on+s390x)**

```
--- FAIL: TestEval (0.44s)
    test_log_scope.go:165: test logs captured to: /tmp/cockroach/_tmp/d76b67a4436d864d4c0c3c8eb62e03f4/logTestEval2491891601
    test_log_scope.go:76: use -show-logs to present logs inline
    --- FAIL: TestEval/opt (0.34s)
        --- FAIL: TestEval/opt/trig (0.00s)
            eval_test.go:46:
                /home/ubuntu/.cache/bazel/_bazel_ubuntu/a136f61a7a0920ebe73e1ee425ce9dee/sandbox/linux-sandbox/230/execroot/com_github_coc\
kroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/sql/sem/eval/eval_test_/eval_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/s\
em/eval/testdata/eval/trig:36:
                  eval
                  atan(1.0)

                expected:
                  0.7853981633974483

                found:
                  0.7853981633974484
    --- FAIL: TestEval/no-opt (0.10s)
        --- FAIL: TestEval/no-opt/trig (0.00s)
            eval_test.go:46:
                /home/ubuntu/.cache/bazel/_bazel_ubuntu/a136f61a7a0920ebe73e1ee425ce9dee/sandbox/linux-sandbox/230/execroot/com_github_coc\
kroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/sql/sem/eval/eval_test_/eval_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/s\
em/eval/testdata/eval/trig:36:
                  eval
                  atan(1.0)

                expected:
                  0.7853981633974483

                found:
                  0.7853981633974484
    eval_test.go:90: -- test log scope end --

```

Epic: CRDB-21133

Jira issue: CRDB-50318

--- a/pkg/sql/sem/eval/testdata/eval/trig
+++ b/pkg/sql/sem/eval/testdata/eval/trig
@@ -1,7 +1,12 @@
 # Trig functions
+#
 # In some cases x64 and arm64 may yeild slightly different results due to float
 # operation ordering/fusing differences, as allowed by spec, so some results are
 # checks using round(expr, 10) to ignore any minute float differences.
+#
+# Similarly, s390x can produce slightly different results (e.g. since it has
+# architectural support for some trig functions like atan), so we check rounded
+# results too.
 
 eval
 cos(0)
@@ -34,14 +39,14 @@ asin(1.0)
 1.5707963267948966
 
 eval
-atan(1.0)
+round(atan(1.0), 15)
 ----
-0.7853981633974483
+0.785398163397448
 
 eval
-atan2(1.0, 1.0)
+round(atan2(1.0, 1.0), 15)
 ----
-0.7853981633974483
+0.785398163397448
 
 eval
 cosd(0)
@@ -74,12 +79,12 @@ asind(1.0)
 90.0
 
 eval
-atand(1.0)
+round(atand(1.0), 10)
 ----
 45.0
 
 eval
-atan2d(1.0, 1.0)
+round(atan2d(1.0, 1.0), 10)
 ----
 45.0
 

